browser/profiles/delete_profile_helper_browsertest.cc

```diff
@@ -0,0 +1,86 @@
+#include "chrome/browser/profiles/profiles_state.h"
+#include "chrome/browser/sync/sync_service_factory.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/test/base/in_process_browser_test.h"
+#include "content/public/test/browser_test.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace {
+const char kValidSyncCode[] =
+    "fringe digital begin feed equal output proof cheap "
+    "exotic ill sure question trial squirrel glove celery "
+    "awkward push jelly logic broccoli almost grocery drift";
+}
+
+// Inspired by src/chrome/browser/profiles/delete_profile_helper_browsertest.cc
+class DeleteProfileHelperBrowserTest : public InProcessBrowserTest {
+ public:
+  DeleteProfileHelperBrowserTest() = default;
+  ~DeleteProfileHelperBrowserTest() override = default;
+};
+
+IN_PROC_BROWSER_TEST_F(DeleteProfileHelperBrowserTest,
+                       SyncStoppedForDeletedProfile) {
+  // Create an additional profile.
+  ProfileManager* profile_manager = g_browser_process->profile_manager();
+  base::FilePath profile_path_to_delete =
+      profile_manager->GenerateNextProfileDirectoryPath();
+  Profile& profile_to_delete = profiles::testing::CreateProfileSync(
+      profile_manager, profile_path_to_delete);
+  EXPECT_TRUE(profile_manager->GetProfileAttributesStorage()
+                  .GetProfileAttributesWithPath(profile_path_to_delete));
+  // Set the profile as last-used, so that the callback of
+  // `MaybeScheduleProfileForDeletion()` is called.
+  Browser::Create(Browser::CreateParams(&profile_to_delete, true));
+  profiles::SetLastUsedProfile(profile_path_to_delete.BaseName());
+  // Schedule profile deletion.
+  ProfileKeepAliveAddedWaiter keep_alive_added_waiter(
+      &profile_to_delete, ProfileKeepAliveOrigin::kProfileDeletionProcess);
+
+  syncer::BraveSyncServiceImpl* brave_sync_service =
+      static_cast<syncer::BraveSyncServiceImpl*>(
+          SyncServiceFactory::GetAsSyncServiceImplForProfileForTesting(
+              &profile_to_delete));
+
+  // Set the sync code to ensure it will be cleared on profile deletion
+  EXPECT_TRUE(brave_sync_service->SetSyncCode(kValidSyncCode));
+
+  brave_sync::Prefs brave_sync_prefs(profile_to_delete.GetPrefs());
+  bool failed_to_decrypt;
+  std::string seed = brave_sync_prefs.GetSeed(&failed_to_decrypt);
+  ASSERT_FALSE(failed_to_decrypt);
+  EXPECT_FALSE(seed.empty());
+
+  base::RunLoop loop;
+  profile_manager->GetDeleteProfileHelper().MaybeScheduleProfileForDeletion(
+      profile_path_to_delete,
+      base::BindLambdaForTesting([&loop](Profile* profile) { loop.Quit(); }),
+      ProfileMetrics::DELETE_PROFILE_PRIMARY_ACCOUNT_NOT_ALLOWED);
+
+  // Ensure that we've invoked BraveSyncServiceImpl::StopAndClear() from
+  // DisableSyncForProfileDeletion at delete_profile_helper.cc, so
+  // now seed should be empty
+  seed = brave_sync_prefs.GetSeed(&failed_to_decrypt);
+  ASSERT_FALSE(failed_to_decrypt);
+  EXPECT_TRUE(seed.empty());
```

Comment:
Shouldn't these be checked after loop.Run() so the test will be more stable since `MaybeScheduleProfileForDeletion` is async?

Reply:
indeed, check after loop would be better, thanks for finding, fixed at 
https://github.com/brave/brave-core/pull/24988/commits/8a5fa75b39c49f8ce53a6e31adf5200406b8d900#diff-3bcc7592f744cd83ffad9eaf185de467750fae6f5c2f17506f4b68a4f432840eR85